assert/opt/opt_test.go

```diff
@@ -236,16 +237,17 @@ func TestPathField(t *testing.T) {
 	filter := PathField(nodeValue{}, "Value")
 	matches := matchPaths(fixture, filter)
 	expected := []string{
-		"{opt.node}.Value.Value",
 		"{opt.node}.Children[0].Value.Value",
-		"{opt.node}.Children[1].Value.Value",
-		"{opt.node}.Children[2].Value.Value",
-		"{opt.node}.Children[2].Ref.Value.Value",
 		"{opt.node}.Children[0].Value.Value",
 		"{opt.node}.Children[1].Value.Value",
-		"{opt.node}.Children[2].Value.Value",
+		"{opt.node}.Children[1].Value.Value",
+		"{opt.node}.Children[2].Ref.Value.Value",
 		"{opt.node}.Children[2].Ref.Value.Value",
+		"{opt.node}.Children[2].Value.Value",
+		"{opt.node}.Children[2].Value.Value",
+		"{opt.node}.Value.Value",
 	}
+	sort.Strings(matches)
```

Comment:
I "fixed" this test, pusing a temporary commit), but not sure if this is correct; 

Looks like results are from matchPaths() are returned in random
order; this sorts the results before comparing, but I'm not sure
if order is important (in which case this could actually be a bug)